packit_service/service/api/projects.py

```diff
@@ -0,0 +1,150 @@
+    def get(self):
+        """List all GitProjects"""
+
+        result = []
+        first, last = indices()
+
+        projects_list = GitProjectModel.get_projects(first, last)
+        for project in projects_list:
+            project_info = {
+                "namespace": project.namespace,
+                "repo_name": project.repo_name,
+                "project_url": project.project_url,
+                "prs_handled": len(project.pull_requests),
+                "branches_handled": len(project.branches),
+                "releases_handled": len(project.releases),
+                "issues_handled": len(project.issues),
+            }
+            result.append(project_info)
+
+        resp = make_response(dumps(result), HTTPStatus.PARTIAL_CONTENT)
+        resp.headers["Content-Range"] = f"git-projects {first + 1}-{last}/*"
+        resp.headers["Content-Type"] = "application/json"
+        resp.headers["Access-Control-Allow-Origin"] = "*"
+        return resp
+
+
+@ns.route("/<forge>/<namespace>/<repo_name>/prs")
+@ns.param("forge", "Git Forge")
+@ns.param("namespace", "Namespace")
+@ns.param("repo_name", "Repo Name")
+class ProjectsPRs(Resource):
+    @ns.expect(pagination_arguments)
+    @ns.response(
+        HTTPStatus.PARTIAL_CONTENT, "Project PRs handled by Packit Service follow"
+    )
+    def get(self, forge, namespace, repo_name):
+        """List PRs"""
+
+        result = []
+        first, last = indices()
+
+        pr_list = GitProjectModel.get_project_prs(
+            first, last, forge, namespace, repo_name
+        )
+        if not pr_list:
+            return ([], HTTPStatus.OK)
+        for pr in pr_list:
+            pr_info = {
+                "pr_id": pr.pr_id,
+                "builds": [],
+                "tests": [],
+            }
+            copr_builds = []
+            test_runs = []
+            for build in pr.get_copr_builds():
+                build_info = {
+                    "build_id": build.build_id,
+                    "chroot": build.target,
+                    "status": build.status,
+                    "web_url": build.web_url,
+                }
+                copr_builds.append(build_info)
+            pr_info["builds"] = copr_builds
+            for test_run in pr.get_test_runs():
+                test_info = {
+                    "pipeline_id": test_run.pipeline_id,
+                    "chroot": test_run.target,
+                    "status": str(test_run.status),
+                    "web_url": test_run.web_url,
+                }
+                test_runs.append(test_info)
+            pr_info["tests"] = test_runs
+
+            result.append(pr_info)
+
+        resp = make_response(dumps(result), HTTPStatus.PARTIAL_CONTENT)
+        resp.headers["Content-Range"] = f"git-project-prs {first + 1}-{last}/*"
+        resp.headers["Content-Type"] = "application/json"
+        resp.headers["Access-Control-Allow-Origin"] = "*"
+        return resp if resp else ([], HTTPStatus.OK)
```

Comment:
I think the `if`-part is not needed here, if there are no PRs, this method will return earlier.